bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -74,8 +74,23 @@ async def reapply_infraction(
             return
 
         # Allowing mod log since this is a passive action that should be logged.
-        await apply_coro
-        log.info(f"Re-applied {infraction['type']} to user {infraction['user']} upon rejoining.")
+        try:
+            await apply_coro
+        except discord.NotFound:
+            # When user joined and then right after this left again before action completed, this can't add roles
+            log.info(f"Can't reapply {infraction['type']} to user {infraction['user']} because user left again.")
+        except discord.HTTPException as e:
+            if e.code == 10007:
+                log.info(f"Can't reapply {infraction['type']} to user {infraction['user']} because user left again.")
```

Comment:
The message is coupled too closely to its sole use in `on_member_join`. However, this function isn't technically restricted to only being used there.
```suggestion
                log.info(f"Can't reapply {infraction['type']} to user {infraction['user']} because user the guild.")
```